pkg/ebpf/c/tracee.bpf.c

```diff
@@ -557,25 +577,31 @@ int tracepoint__sched__sched_process_fork(struct bpf_raw_tracepoint_args *ctx)
     if (!should_trace(&p))
         return 0;
 
-    // follow every pid that passed the should_trace() checks (used by the follow filter)
+    // Always follow every pid that passed the should_trace() checks (follow filter)
     c_proc_info->follow_in_scopes = p.task_info->matched_scopes;
 
-    if (should_submit(SCHED_PROCESS_FORK, p.event) || p.config->options & OPT_PROCESS_INFO) {
-        int parent_ns_pid = get_task_ns_pid(parent);
-        int parent_ns_tgid = get_task_ns_tgid(parent);
-        int child_ns_pid = get_task_ns_pid(child);
-        int child_ns_tgid = get_task_ns_tgid(child);
```

Comment:
I can move the following:

```C
        u64 parent_start_time = get_task_start_time(parent);
        int parent_tid = get_task_host_pid(parent);
        int parent_ns_pid = get_task_ns_tgid(parent);
        int parent_ns_tid = get_task_ns_pid(parent);
        // If the "real parent" is a thread, we need to get the leader's info
        u64 leader_start_time = get_task_start_time(leader);
        int leader_pid = get_task_host_tgid(leader);
        int leader_tid = get_task_host_pid(leader);
        int leader_ns_pid = get_task_ns_tgid(leader);
        int leader_ns_tid = get_task_ns_pid(leader);
```

to after the should_submit check. This helps saving a few cycles as well (mostly because of the bpf helpers overhead I believe, cause the instructions per-se would be irrelevant).